core/src/components/refresher/refresher.tsx

```diff
@@ -443,34 +443,28 @@ export class Refresher implements ComponentInterface {
       console.error('Make sure you use: <ion-refresher slot="fixed">');
       return;
     }
-
     const contentEl = this.el.closest(ION_CONTENT_ELEMENT_SELECTOR);
     if (!contentEl) {
       printIonContentErrorMsg(this.el);
       return;
     }
-
+    /**
+     * Waits for the content to be ready before querying the scroll
+     * or the background content element.
+     */
+    await contentEl.componentOnReady();
     const customScrollTarget = contentEl.querySelector(ION_CONTENT_CLASS_SELECTOR);
-
     /**
      * Query the custom scroll target (if available), first. In refresher implementations,
      * the ion-refresher element will always be a direct child of ion-content (slot="fixed"). By
      * querying the custom scroll target first and falling back to the ion-content element,
      * the correct scroll element will be returned by the implementation.
      */
     this.scrollEl = await getScrollElement(customScrollTarget ?? contentEl);
-
     /**
-     * Query the host `ion-content` directly (if it is available), to use its
-     * inner #background-content has the target. Otherwise fallback to the
-     * custom scroll target host.
-     *
-     * This makes it so that implementers do not need to re-create the background content
-     * element and styles.
+     * Query the background content element from the host ion-content element directly.
      */
-    this.backgroundContentEl = getElementRoot(contentEl ?? customScrollTarget).querySelector(
-      '#background-content'
-    ) as HTMLElement;
+    this.backgroundContentEl = await contentEl.getBackgroundElement();
```

Comment:
Do we still need to account for custom scroll targets that create their own `#background-content` element?

Reply:
Nope. Technically they never could based on how I originally implemented it. We have a check above this that requires an `ion-content` element or prints an error and returns early. The check for `contentEl ?? customScrollTarget` would always resolve the `ion-content`. 

In hindsight, this is how we expected it to behave. We did not want developers ejecting from `ion-content` and also needing to provide the background content element. It just got lost in translation while trying to explore the alternative bug that overlapped with much of that work.